src/types.rs

```diff
@@ -28,10 +32,28 @@ impl NpyDataType {
     pub(crate) fn from_i32(npy_t: i32) -> Self {
         match npy_t {
             x if x == NPY_TYPES::NPY_BOOL as i32 => NpyDataType::Bool,
+            x if x == NPY_TYPES::NPY_BYTE as i32 => NpyDataType::Int8,
+            x if x == NPY_TYPES::NPY_SHORT as i32 => NpyDataType::Int16,
             x if x == NPY_TYPES::NPY_INT as i32 => NpyDataType::Int32,
-            x if x == NPY_TYPES::NPY_LONG as i32 => NpyDataType::Int64,
+            x if x == NPY_TYPES::NPY_LONG as i32 => {
+                if cfg!(any(windows, target_pointer_width = "32")) {
+                    NpyDataType::Int32
+                } else {
+                    NpyDataType::Int64
+                }
+            }
```

Comment:
This should not be a windows issue.

> NPY_LONG
> Equivalent to either NPY_INT or NPY_LONGLONG, depending on the platform.

`long` type in C is platform dependent. 
`std::os::raw::c_long` handle this difference in Rust, and it must be same as NumPy.
https://doc.rust-lang.org/std/os/raw/type.c_long.html

It will be better to check by the `std::mem::size_of` `c_long` and `i32`.

Reply:
But [std::os::raw](https://github.com/rust-lang/rust/blob/master/src/libstd/os/raw/mod.rs) also uses `#[cfg(any(target_pointer_width = "32", windows))]` so now I feel current implementation is enough :confused:

Reply:
I see. That's right.